verifier/src/main/scala/org/apache/celeborn/verifier/scheduler/Scheduler.scala

```diff
@@ -0,0 +1,289 @@
+                    actionCandidates(Random.nextInt(actionCandidates.size))
+                  } else {
+                    logInfo(s"Plan loop: $planLoopCount.")
+                    actions(actionIndex)
+                  }
+                val operations = actionToExecute.generateOperations(schedulerContext)
+                logDebug(
+                  s"Current action: $actionToExecute, operations: ${operations.map(_.toString)}.")
+                val permissionToExecute =
+                  if (operations.isEmpty) {
+                    false
+                  } else {
+                    val deducedContext = actionToExecute.deduce(schedulerContext, operations)
+                    val permissionToRun = checker.validate(deducedContext)
+                    logDebug(s"Current context: ${schedulerContext.toString}  => deduced context: ${deducedContext.toString}, " +
+                      s"current action executes $permissionToRun.")
+                    permissionToRun
+                  }
+                if (permissionToExecute) {
+                  def executeOperationSync(operation: Operation): Unit = {
+                    val executeOperationFuture =
+                      runnerInfos.get(operation.actionTarget.identity).rpcEndpointRef
+                        .ask[ExecuteOperationResponse](ExecuteOperation(operation))
+                    val executeOperationResponse =
+                      ThreadUtils.awaitReady(executeOperationFuture, Duration.Inf).value.get.get
+                    if (executeOperationResponse.statusCode != StatusCode.SUCCESS) {
+                      throw new PlanExecutionException("Operation failed, plan execution abort.")
+                    }
+                    operation.updateSchedulerContext(schedulerContext)
+                    Thread.sleep(operation.interval)
+                  }
+                  operations.foreach(executeOperationSync)
+                  Thread.sleep(actionToExecute.interval)
+                }
+                if (trigger.sequence) {
+                  actionIndex = actionIndex + 1
+                  if (actionIndex >= actionCount - 1) {
+                    planLoopCount = planLoopCount + 1
+                    if (planLoopCount > trigger.repeat) {
+                      logInfo("Plan execution completed, cleaning verification plan.")
+                      stopPlanFlag.set(true)
+                      runningPlan.set(null)
+                    } else {
+                      actionIndex = 0
+                    }
+                  }
+                }
+                Thread.sleep(trigger.interval.getInterval)
+              }
+            }
+          }
+        } catch {
+          case t: Throwable =>
+            try {
+              logError("Plan executed with exception, cleaning verification plan.", t)
+              planExecuteTask = planExecutor.submit(new PlanLoop)
+              stopPlanFlag.set(true)
+              runningPlan.set(null)
```

Comment:
QQ: Why are we resubmitting the `PlanLoop` ? (given we have set  `stopPlanFlag` to `true` anyway)